### PR TITLE
Final round of clipboard cleanup

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Following logic should be significantly easier now.

- Move methods inline in ComDataObjectAdapter
- Make methods static and pass in state
- Move a few methods inlinein DataObject
- Use Span to save filelist
- Use HGLOBAL terms

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9205)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
>Make methods static and pass in state

what is the reason for this change?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataObject.ComDataObjectAdapter.cs [hunk 1]
```diff
@@ -189,9 +215,9 @@ static unsafe string ReadUtf8StringFromHGLOBAL(nint hglobal)
 
             return data;
 
-            // Currently this only supports Bitmap
-            static object? GetDataFromOleOther(string format, IComDataObject dataObject)
+            static object? TryGetBitmapData(IComDataObject dataObject, string format)
             {
+                // Currently this only supports Bitmap.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataObject.ComDataObjectAdapter.cs:220]: Function name makes this comment redundant.

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by JeremyKuhne** [comment]
> what is the reason for this change?

To make instantly clear what the dependencies are- only the parameters. Otherwise you need to know what state the instance has.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataObject.ComDataObjectAdapter.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataObject.ComDataObjectAdapter.cs:220]: I'll remove it in the BF change.

**Pull request merged**